Services/Concurrency/Timer.cs

```diff
@@ -57,14 +62,32 @@ public ITimer StartIn(TimeSpan delay)
                 throw new TimerNotInitializedException();
             }
 
+            this.stopped = false;
             this.timer.Change((int) delay.TotalMilliseconds, this.frequency);
             return this;
         }
 
+        public void Pause()
+        {
+            if (!this.stopped)
+            {
+                this.timer?.Change(Timeout.Infinite, this.frequency);
+            }
+        }
+
+        public void Resume()
+        {
+            if (!this.stopped)
+            {
+                this.timer?.Change(this.frequency, this.frequency);
```

Comment:
agreed, I had a similar concern about this way of using the timers. For example, in a scenario where messages should be sent every 10 seconds, if sending a message takes 2 seconds, the application will end up sending messages every 12 seconds. We should track the time passed since Pause and Resume subtracting the time passed. However, there are tricky edge cases to consider. I'll see if I can fix this in this PR, otherwise create a separate task

Reply:
fixed